internal/app/machined/pkg/runtime/v1alpha1/platform/platform.go

```diff
@@ -113,7 +113,11 @@ func newPlatform(platform string) (p runtime.Platform, err error) {
 	case "hcloud":
 		p = &hcloud.Hcloud{}
 	case constants.PlatformMetal:
-		p = &metal.Metal{}
+		_, metalAgentCheckErr := os.Stat("/usr/local/etc/is-metal-agent")
```

Comment:
I know it's used only once, but it'd be nice to move it out to `pkg/machinery/constants` for consistency

Reply:
done